### ISSUE TITLE
TypeScript Types

### ISSUE THREAD

**Issue by carlosbaraza** [ircam-ismm/node-web-audio-api#32]
Thank you for the great work. I was wondering if types could be added to the library. napi-rs seems to support generating a declaration file.

**Comment by carlosbaraza** [comment]
In case it works for anyone, this is the declaration file I am currently using:

```typescript
// Referencing the default lib web api typings
/// <reference lib="dom" />

declare module 'node-web-audio-api' {
  export import AudioContext = globalThis.AudioContext;
  export import OfflineAudioContext = globalThis.OfflineAudioContext;
  export import AudioBuffer = globalThis.AudioBuffer;
  export import PeriodicWave = globalThis.PeriodicWave;
  export import MediaStreamAudioSourceNode = globalThis.MediaStreamAudioSourceNode;
  export import AnalyserNode = globalThis.AnalyserNode;
  export import AudioBufferSourceNode = globalThis.AudioBufferSourceNode;
  export import BiquadFilterNode = globalThis.BiquadFilterNode;
  export import ChannelMergerNode = globalThis.ChannelMergerNode;
  export import ChannelSplitterNode = globalThis.ChannelSplitterNode;
  export import ConstantSourceNode = globalThis.ConstantSourceNode;
  export import ConvolverNode = globalThis.ConvolverNode;
  export import DelayNode = globalThis.DelayNode;
  export import DynamicsCompressorNode = globalThis.DynamicsCompressorNode;
  export import GainNode = globalThis.GainNode;
  export import IIRFilterNode = globalThis.IIRFilterNode;
  export import OscillatorNode = globalThis.OscillatorNode;
  export import PannerNode = globalThis.PannerNode;
  export import StereoPannerNode = globalThis.StereoPannerNode;
  export import WaveShaperNode = globalThis.WaveShaperNode;
}
```

The biggest drawback is that some properties or outputs might be different than the browser Web Audio API.

**Comment by carlosbaraza** [comment]
I tried to fork the library and modify it, but I couldn't really figure out how to make napi-rs output types. But this simple workaround seems to work well for me. Feel free to add it to the library. Better this than nothing, IMO.

**Comment by carlosbaraza** [comment]
Just submitted a PR.

For reference (pun intended), this is the explanation of GPT-4 of the reference directive, much better than what I can explain:

```
How does the "reference" thing works?
ChatGPT
The /// <reference ... /> directive is a TypeScript triple-slash directive. It is used as a way to declare dependencies between files or to reference type declarations that aren't automatically discoverable by TypeScript's normal module resolution. These directives are only used in the absence of proper module systems or when working with global scripts.

Here's a breakdown of how it works:

Referencing Type Declarations:
When working with global scripts (i.e., scripts that aren't modules and don't use import or export), TypeScript doesn't automatically include all definition files (*.d.ts). The triple-slash reference directive instructs the compiler to include an additional file as part of the compilation process.

typescript
Copy code
/// <reference path="path-to-file.d.ts" />
Library References:
The lib option in the directive (as seen in your provided code) is used to include built-in type declarations from the TypeScript standard library.

typescript
Copy code
/// <reference lib="dom" />
This includes the type declarations for the Web DOM API, which makes DOM-specific types and globals (like window, document, or HTMLElement) available in the file. TypeScript has several built-in libraries, such as es2015, esnext, dom, etc., which you can reference using this directive.

How it is Used:
When the TypeScript compiler encounters a triple-slash reference directive, it will follow the reference and include the mentioned file as part of the compilation. This is especially useful when you want to explicitly set the order of file compilation or when TypeScript can't automatically find the necessary type declarations.

Modern Usage:
In modern TypeScript projects that use modules (i.e., they use import and export), triple-slash directives are less common because TypeScript can automatically discover type declarations through its standard module resolution process. However, in certain edge cases or configurations, or when dealing with global scripts, these directives can still be useful.

In your code, the line /// <reference lib="dom" /> is telling TypeScript to include the built-in type declarations for the Web DOM API, allowing the subsequent code to reference types like globalThis.AudioContext without any type errors.
```

**Comment by b-ma** [comment]
Thanks for the PR and explanations!

Will close the issue once released

**Comment by b-ma** [comment]
released in v0.13.0
